src/less/star-rating/base/star-rating.less

```diff
@@ -10,3 +10,34 @@ svg.star-rating {
     vertical-align: middle;
     width: 64px;
 }
+
+div.star-rating svg.star-rating-icon {
+    fill: none;
+    height: 12px;
+    width: 12px;
+}
+
+/* each star starts unfilled */
+div.star-rating svg.star-rating-icon use.star-rating-icon__star-full {
+    .fill-token(color-star-rating-icon-unfilled-foreground, color-star-rating-icon-unfilled-foreground);
+}
+
+/* full stars */
+div.star-rating[data-stars^="1"] svg:nth-child(-n + 1) use.star-rating-icon__star-full,
+div.star-rating[data-stars^="2"] svg:nth-child(-n + 2) use.star-rating-icon__star-full,
+div.star-rating[data-stars^="3"] svg:nth-child(-n + 3) use.star-rating-icon__star-full,
+div.star-rating[data-stars^="4"] svg:nth-child(-n + 4) use.star-rating-icon__star-full,
+div.star-rating[data-stars^="5"] svg:nth-child(-n + 5) use.star-rating-icon__star-full {
+    .fill-token(color-star-rating-icon-filled-foreground, color-star-rating-icon-filled-foreground);
+}
+
+/* half stars */
+div.star-rating[data-stars="0-5"] svg:nth-child(1) use.star-rating-icon__star-half,
```

Comment:
Just looking at this isn't there a way to do this without multiple uses and to color the paths instead?
If we combine the stars like this:
<symbol id="star-rating-combo">
   <path class="star-rating">
   <path class="star-rating-half">
   <path class="star-rating-half">
</symbol>

And then do the same things on the path directly it should work. The only issue might be shadow dom problems

Another solve would be to use CSS variables in the root and they would cascade down.

Reply:
Hmm, if I'm understanding your suggestion accurately, it would mean we will need to use `<symbol>`s repetitively. Since `<symbol>`s are templates I don't think they are supposed to be used that way. They are defined once and are used by <`use>`. That solution would be extremely verbose, if even possible.

Using `<path>`s would mean we'd have to move/define the icons inline which would make usage quite a bit more complicated and potentially more easily breakable by implementors.

In the `SVG` world, having multiple `<use>` tags in an `<svg>` is standard and common (https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use); so I'm wondering whether it would make more sense to actually extend core ui to support multiple `<use>` tags inside an `<svg>` instead.

> Another solve would be to use CSS variables in the root and they would cascade down.

I didn't quite understand this. How would this remove multiple `<use>` tags?

If the barrier is too large to support multiple `<use>` tags right now, I'll poke around for another solution.

Reply:
> Hmm, if I'm understanding your suggestion accurately, it would mean we will need to use `<symbol>`s repetitively. Since `<symbol>`s are templates I don't think they are supposed to be used that way. They are defined once and are used by <`use>`. That solution would be extremely verbose, if even possible.

So no, it's not supposed to use symbols repetitively. It would not use more than it's currently being used. 
> 
> Using `<path>`s would mean we'd have to move/define the icons inline which would make usage quite a bit more complicated and potentially more easily breakable by implementors.
> 
> In the `SVG` world, having multiple `<use>` tags in an `<svg>` is standard and common (https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use); so I'm wondering whether it would make more sense to actually extend core ui to support multiple `<use>` tags inside an `<svg>` instead.
Possibly, but I don't think it's necessary since I think there are other ways around it. Using multiple uses is mostly for combining images and not necessarily for hiding one and showing another imo. This is mostly because our icons are static and we don't combine the same shapes.
> 
> > Another solve would be to use CSS variables in the root and they would cascade down.
> 
> I didn't quite understand this. How would this remove multiple `<use>` tags?
Answered below
> 
> If the barrier is too large to support multiple `<use>` tags right now, I'll poke around for another solution.

So here's what I propose:

In icons.svg we have a symbol like this:

```
<symbol id="icon-star-combined" viewBox="...">
   <path class="star" d="..."/>
   <path class="star-half" d="..."/>
   <path class="star-half" d="..."/>
</symbol>
```

And in your template for star rating you would simply do:

```
<svg> 
   <use xlink:href="#icon-star-combined">
</svg>
```

And your css instead of targeting the `use`, target the `star` or `star-half` classes. 
If that is not possible because of shadow dom issues then this would work (and this is what we do for dark mode variables):
```
<symbol id="icon-star-combined" viewBox="...">
   <path style="opacity(var(--star-selected, 0.25))" d="..."/>
   <path style="opacity(var(--star-half-selected, 0))" d="..."/>
   <path style="opacity(var(--star-half-selected, 0))" d="..."/>
</symbol>
```
And then in the CSS for the root SVG you could set the `--star-selected: 1` and `--star-half-selected: 1` to shade it correctly.

Reply:
Ok, I'll try that but this approach will likely compromise future micro-animation treatments. I guess we'll deal with that when we get to it.

Reply:
Sure, I think the animation is great to have but it's not in the design spec. 
If and when it comes we can revisit it

Reply:
FYI, one of the nice things about the current/existing method that we have in place, is that it works without CSS:

![Screen Shot 2022-03-29 at 3 16 27 PM](https://user-images.githubusercontent.com/38065/160716124-1a7bddd4-6a50-4b8f-8fd8-90bf4742b646.png)

Reply:
Do you mean this current approach or the one with just svgs?
Again if we do have the svgs only approach doing a static one on ebayui is fine as well. We just have to programmatically import then rating we want to show.

Reply:
Oops, I meant the current one in production:

https://ebay.github.io/skin/#star-rating

Reply:
Happy to proceed, just something to be aware of and one of the reasons why it's probably good to keep around the production approach too, as it fully supports progressive enhancement.

If we are getting stalled on this one, then maybe we shelve it so that we can move onto the interactive star buttons in v14? Or is that dependent on nailing down this PR?

Reply:
> Or is that dependent on nailing down this PR?

I was initially doing this somewhat in tandem with the interactive one, and with this change in approach the interactive one will likely need a change (trying to avoid context switching for now to check) in direction too. 

I think at least on some level, a single star approach would be a prerequisite to the interactive version. So even if we kill this pr, we'll need to do something similar for the interactive.

> If we are getting stalled on this one

I'm not stalled; just a sizable change in approach late in the game; working on it now...